File: android/src/main/java/ru/innim/flutter_login_vk/FlutterError.java
package ru.innim.flutter_login_vk;

import com.vk.sdk.api.VKError;

public class FlutterError {
    String code;
    String message;
    Object details;

    public static FlutterError invalidArgs(String message, Object details) {
        return new FlutterError("INVALID_ARGS", message, details);
    }

    public static FlutterError invalidResult(String message, Object details) {
        return new FlutterError("INVALID_RESULT", message, details);
    }

    public static FlutterError apiUnavailable(String message, Object details) {
        return new FlutterError("API_UNAVAILABLE", message, details);
    }

    public static FlutterError interrupted(String message, Object details) {
        return new FlutterError("INTERRUPTED", message, details);
    }

    public static FlutterError noConnection(String message, Object details) {
        return new FlutterError("NO_CONNECTION", message, details);
    }

    public static FlutterError apiError(String message, VKError error) {
        return new FlutterError("API_ERROR", message, Results.error(error));
    }

    public FlutterError(String code, String message, Object details) {
        this.code = code;
        this.message = message;
        this.details = details;
    }
}

File: android/src/main/kotlin/ru/innim/flutter_login_vk/FlutterError.kt
package ru.innim.flutter_login_vk

class FlutterError(var code: String, var message: String?, var details: Any?) {
    companion object {
        fun invalidArgs(message: String, details: Any?): FlutterError {
            return FlutterError("INVALID_ARGS", message, details)
        }

        fun invalidResult(message: String, details: Any? = null): FlutterError {
            return FlutterError("INVALID_RESULT", message, details)
        }

        fun apiUnavailable(message: String, details: Any? = null): FlutterError {
            return FlutterError("API_UNAVAILABLE", message, details)
        }

        fun interrupted(message: String, details: Any? = null): FlutterError {
            return FlutterError("INTERRUPTED", message, details)
        }

        fun noConnection(message: String, details: Any? = null): FlutterError {
            return FlutterError("NO_CONNECTION", message, details)
        }

        fun apiError(message: String, error: VKError): FlutterError {
            return FlutterError("API_ERROR", message, Results.error(error))
        }
    }
}
